### PR TITLE
Compatibility fixes

### PR DESCRIPTION
**Pull request by vbekiaris**
Fixes for `PostJoinMapOperation` upgrade compatibility from 3.9 to 3.10.
See individual commit messages for additional info.

### CODE REVIEW

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/PostJoinMapOperation.java [hunk 1]
```diff
@@ -190,9 +191,9 @@ protected void writeInternal(ObjectDataOutput out) throws IOException {
     protected void readInternal(ObjectDataInput in) throws IOException {
         super.readInternal(in);
         // RU_COMPAT_39
-        // Version 3.9.x still sends index info's because PostJoinMapOperation was not marked Versioned
+        // Version 3.9.x still sends index info's from EE because PostJoinMapOperation was not marked Versioned
         Version inputversion = in.getVersion();
-        if (inputversion.isUnknown() || inputversion.isEqualTo(V3_9)) {
+        if ((inputversion.isUnknown() || inputversion.isEqualTo(V3_9)) && getBuildInfo().isEnterprise()) {
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/PostJoinMapOperation.java]: Is this the reason for the OOM in the RU test?
(and I think I also saw it in a real-life test from Danny)

Comment (vbekiaris) [reply hazelcast/src/main/java/com/hazelcast/map/impl/operation/PostJoinMapOperation.java]: Yes, this usually manifests either as OOME (I think on the 3.9.x side) or as malformed bytes sequence when attempting to read UTF string.

**Comment by vbekiaris** [comment]
Test failure #12376 
https://hazelcast-l337.ci.cloudbees.com/job/new-lab-fast-pr/13941/testReport/junit/com.hazelcast.map/EvictionTest/testTTL_prolongationAfterNonTTLUpdate_Quick/

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

**Pull request merged**